atlas: remove

matthewbauer opened the pull request:
atlas is broken and can apparently be removed.

Fixes #49594

matthewbauer committed:
atlas: remove

atlas is broken and can apparently be removed.

Fixes #49594

markuskowa commented:
`blas` and `liblapack` may be replaced by `openblasCompat` here but this step requires further testing.
CC @symphorien @veprbl

veprbl commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21203,13 +21191,7 @@ with pkgs;
-  # without atlas. Etc.
   liblapack = callPackage ../development/libraries/science/math/liblapack {};
-  liblapackWithoutAtlas = liblapackWithAtlas.override { atlas = null; };
-  liblapackWithAtlas = liblapack;
```
`liblapackWithoutAtlas` would need to go to `aliases.nix`

matthewbauer committed:
liblapack: add compat aliases

matthewbauer commented:
@grahamcofborg build petsc openmodelica shogun giac openblas

GrahamcOfBorg commented:
<!--REQUEST_ID=8e6ad5f5-c0c5-48b9-a19b-0a4ae12d32ac-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=4bbca2f0-9f4a-4b60-9721-f15f5a09d5ff)

Attempted: petsc, shogun, openblas

The following builds were skipped because they don't evaluate on x86_64-darwin: openmodelica, giac

<details><summary>Partial log (click to expand)</summary><p>

```
make PETSC_DIR=/nix/store/mzlyphzrjqs1g6yjmi6dpbxblfi7spq2-petsc-3.8.4 PETSC_ARCH="" test
====================================
/nix/store/f5aw0jl41r3nydav2mprvd2zyb76hg92-gnumake-4.2.1/bin/make  --no-print-directory PETSC_ARCH=arch-darwin-c-debug PETSC_DIR=/private/tmp/nix-build-petsc-3.8.4.drv-0/petsc-3.8.4 mpi4py-install petsc4py-install
make[1]: Nothing to be done for 'mpi4py-install'.
make[1]: Nothing to be done for 'petsc4py-install'.
post-installation fixup
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/mzlyphzrjqs1g6yjmi6dpbxblfi7spq2-petsc-3.8.4/lib  /nix/store/mzlyphzrjqs1g6yjmi6dpbxblfi7spq2-petsc-3.8.4/bin
patching script interpreter paths in /nix/store/mzlyphzrjqs1g6yjmi6dpbxblfi7spq2-petsc-3.8.4
error: build of '/nix/store/ij8pl9nda6kfl48s1d4g8x2iw15nzv0b-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8e6ad5f5-c0c5-48b9-a19b-0a4ae12d32ac-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=75af20b0-4b17-43f9-b258-b573964ae20d)

Attempted: petsc, shogun, giac, openblas

The following builds were skipped because they don't evaluate on x86_64-linux: openmodelica

<details><summary>Partial log (click to expand)</summary><p>

```
In file included from /build/source/src/shogun/classifier/svm/NewtonSVM.cpp:19:0:
/build/source/src/shogun/mathematics/lapack.h:37:10: fatal error: cblas.h: No such file or directory
 #include <cblas.h>
          ^~~~~~~~~
compilation terminated.
make[2]: *** [src/shogun/CMakeFiles/libshogun.dir/build.make:1381: src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: src/shogun/CMakeFiles/libshogun.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/3wrvkrv5hg5rjlv8j2zqn0ancwms8gga-shogun-6.0.0.drv' failed with exit code 2
error: build of '/nix/store/3wrvkrv5hg5rjlv8j2zqn0ancwms8gga-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8e6ad5f5-c0c5-48b9-a19b-0a4ae12d32ac-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=a6f97b53-d435-4e08-a5e6-4a85613dd5b9)

Attempted: petsc, shogun, giac, openblas

The following builds were skipped because they don't evaluate on aarch64-linux: openmodelica

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/nhx8pb4n2pilbi2aq2621y4m7g0wy3mm-giac-1.4.9-59/lib  /nix/store/nhx8pb4n2pilbi2aq2621y4m7g0wy3mm-giac-1.4.9-59/bin
patching script interpreter paths in /nix/store/nhx8pb4n2pilbi2aq2621y4m7g0wy3mm-giac-1.4.9-59
/nix/store/nhx8pb4n2pilbi2aq2621y4m7g0wy3mm-giac-1.4.9-59/bin/pgiac: interpreter directive changed from "/usr/bin/perl" to "/nix/store/bzgcpj9fijmlxgjaa72fjys7m3ycijy8-perl-5.28.0/bin/perl"
checking for references to /build in /nix/store/nhx8pb4n2pilbi2aq2621y4m7g0wy3mm-giac-1.4.9-59...
shrinking RPATHs of ELF executables and libraries in /nix/store/k09lcz1hwbind8nmsqxmzj1jcxv7ypq3-giac-1.4.9-59-doc
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/k09lcz1hwbind8nmsqxmzj1jcxv7ypq3-giac-1.4.9-59-doc
checking for references to /build in /nix/store/k09lcz1hwbind8nmsqxmzj1jcxv7ypq3-giac-1.4.9-59-doc...
error: build of '/nix/store/ld2f5bph5s5iv4z0bmdl7nxwy9a5smvs-shogun-6.0.0.drv' failed
```
</p></details>

markuskowa commented:
Using openblas instead of blas seems to work here (I can compile locally).
```
--- a/pkgs/applications/science/machine-learning/shogun/default.nix
+++ b/pkgs/applications/science/machine-learning/shogun/default.nix
@@ -2,7 +2,7 @@
 # data, compression
 , bzip2, curl, hdf5, json_c, lzma, lzo, protobuf, snappy
 # maths
-, blas, eigen, nlopt, lp_solve, colpack
+, openblasCompat, eigen, nlopt, lp_solve, colpack
 # libraries
 , libarchive, liblapack, libxml2
 # extra support
@@ -41,7 +41,7 @@ stdenv.mkDerivation rec {
   CCACHE_DIR=".ccache";
 
   buildInputs = with lib; [
-      blas bzip2 ccache cmake colpack curl ctags eigen hdf5 json_c lp_solve lzma lzo
+      openblasCompat bzip2 ccache cmake colpack curl ctags eigen hdf5 json_c lp_solve lzma lzo
       protobuf nlopt snappy swig (libarchive.dev) liblapack libxml2
     ]
     ++ optionals (pythonSupport) (with pythonPackages; [ python ply numpy ])
```

matthewbauer committed:
shogun: use openblas

shogun needs the cblas.h header file to be available. It used to get
it from liblapack, but that is not available anymore. Instead we can
use openblasCompat to get it.

matthewbauer committed:
aliases: remove liblapackWithAtlas

This doesn’t make any sense to keep - we don’t have a liblapack with
atlas anymore.

GrahamcOfBorg commented:
<!--REQUEST_ID=77aea0c1-da44-465c-9a14-939d7dfa57c7-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=162eb8de-2513-4247-91a6-eb6f1b6adb2a)

Attempted: scs, gmsh, csdp, scalapack

The following builds were skipped because they don't evaluate on x86_64-darwin: magma, mxnet

<details><summary>Partial log (click to expand)</summary><p>

```
         91 - xssyevr (Failed)
         92 - xdsyevr (Failed)
         93 - xcheevr (Failed)
         94 - xzheevr (Failed)
         95 - xshseqr (Failed)
         96 - xdhseqr (Failed)
Errors while running CTest
make: *** [Makefile:130: test] Error 8
builder for '/nix/store/8yj4dl5z3f506q37zbqifyjqscqdp89j-scalapack-2.0.2.drv' failed with exit code 2
error: build of '/nix/store/099w05ms0hg2yiwwai676f9vyh8224dk-gmsh-4.0.2.drv', '/nix/store/8yj4dl5z3f506q37zbqifyjqscqdp89j-scalapack-2.0.2.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=93435800-58da-499f-bfa1-75d0772a23c7-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=1f6dd903-1f08-473b-9ce6-d0e55ce06e27)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parallel.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parameter.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o
/build/ccQ1RdXJ.s: Assembler messages:
/build/ccQ1RdXJ.s:3821: Error: unknown mnemonic `pause' -- `pause'
make[2]: *** [src/shogun/CMakeFiles/libshogun.dir/build.make:1030: src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: src/shogun/CMakeFiles/libshogun.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/1p9mbllmfzgvl0p8s82kyc2hc2f7p8vl-shogun-6.0.0.drv' failed with exit code 2
error: build of '/nix/store/1p9mbllmfzgvl0p8s82kyc2hc2f7p8vl-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=17eec1f5-3852-4bc5-a668-758107f5871b-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=75629c01-88ea-4e56-899b-13faf947b27f)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
./Main/ColPackHeaders.h:61:11: fatal error: 'omp.h' file not found
        #include <omp.h>
                 ^~~~~~~
1 error generated.
make[1]: *** [Makefile:1489: SampleDrivers/Basic/color_bipartite_graph_using_BipartiteGraphBicoloringInterface.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-ColPack-1.0.10.drv-0/source'
make: *** [Makefile:859: all] Error 2
builder for '/nix/store/2qdwr8gif8qixqvn5xfh0y6hvqg2q4f1-ColPack-1.0.10.drv' failed with exit code 2
cannot build derivation '/nix/store/bmabvfgglparcmcjcr6ygyl2wvs6843b-shogun-6.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/bmabvfgglparcmcjcr6ygyl2wvs6843b-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=17eec1f5-3852-4bc5-a668-758107f5871b-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=c970d897-432d-41d1-a4c7-24a6b800626d)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parallel.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parameter.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o
/build/ccHhNgMD.s: Assembler messages:
/build/ccHhNgMD.s:3821: Error: unknown mnemonic `pause' -- `pause'
make[2]: *** [src/shogun/CMakeFiles/libshogun.dir/build.make:1030: src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: src/shogun/CMakeFiles/libshogun.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/1p9mbllmfzgvl0p8s82kyc2hc2f7p8vl-shogun-6.0.0.drv' failed with exit code 2
error: build of '/nix/store/1p9mbllmfzgvl0p8s82kyc2hc2f7p8vl-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=93435800-58da-499f-bfa1-75d0772a23c7-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=7ad9a041-fc72-4e7e-80f3-f0d07a423924)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0/lib/cmake/lapack-3.8.0/lapack-config-version.cmake
-- Installing: /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0/lib/pkgconfig/blas.pc
-- Installing: /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0/lib/libblas.a
-- Installing: /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0/lib/liblapack.a
post-installation fixup
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0/lib
patching script interpreter paths in /nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0
cannot build derivation '/nix/store/bmabvfgglparcmcjcr6ygyl2wvs6843b-shogun-6.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/bmabvfgglparcmcjcr6ygyl2wvs6843b-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=93435800-58da-499f-bfa1-75d0772a23c7-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=10f8d902-b707-45ed-a0a2-5e0bbaaa1b69)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/share/shogun/examples/libshogun/classifier_mkl_svmlight_modelselection_bug
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/share/shogun/examples/libshogun/serialization_basic_tests
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/share/shogun/examples/libshogun/classifier_multiclasslinearmachine
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/share/shogun/examples/libshogun/parameter_set_from_parameters
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/lib/python2.7/site-packages/_modshogun.so
shrinking /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/lib/libshogun.so.18.0
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0/lib
patching script interpreter paths in /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0
checking for references to /build in /nix/store/c4xi5qa6jnnz3aj8sjp8rwb3vn7374ds-shogun-6.0.0...
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=77aea0c1-da44-465c-9a14-939d7dfa57c7-->
Timed out, unknown build status on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=028707cd-5aca-4560-8eb6-56d79148b8d8)

Attempted: scs, gmsh, csdp, mxnet, scalapack

The following builds were skipped because they don't evaluate on x86_64-linux: magma

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2/share/man/man1/gmsh.1
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2
shrinking /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2/bin/gmsh
gzipping man pages under /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2/share/man/
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2/bin
patching script interpreter paths in /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2
checking for references to /build in /nix/store/jmsv8761d3ycpwghfmzg56i56rpy6c5a-gmsh-4.0.2...
error: build of '/nix/store/zcllzj9ifvfgf4307i2gmhr7inf2aqkg-mxnet-1.2.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=17eec1f5-3852-4bc5-a668-758107f5871b-->
Timed out, unknown build status on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=fbcab600-6a74-4d48-8fae-7887892a336b)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
[ 63%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/FactorGraphModel.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/FactorType.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/GEMPLP.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/GraphCut.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/HMSVMModel.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/HashedMultilabelModel.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/HierarchicalMultilabelModel.cpp.o
[ 64%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/structure/IntronList.cpp.o
building of '/nix/store/cxm6pncgw12yrs2cr5rp98w5p4f9hjxs-shogun-6.0.0.drv' timed out after 3600 seconds
error: build of '/nix/store/cxm6pncgw12yrs2cr5rp98w5p4f9hjxs-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=77aea0c1-da44-465c-9a14-939d7dfa57c7-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=d3849b06-268e-40d2-bcd1-9860847f18f4)

Attempted: scs, gmsh, csdp, mxnet, scalapack

The following builds were skipped because they don't evaluate on aarch64-linux: magma

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/by5p7vlw6p1hsixyliy85dim5r8xqzq5-conduit-1.3.1.drv': 14 dependencies couldn't be built
cannot build derivation '/nix/store/q3pv07v48jyh8qzgpm1r8wjv6yc5va57-tls-1.4.1.drv': 17 dependencies couldn't be built
cannot build derivation '/nix/store/x94qhdgjjn6pi5zhcnbxp6csmndvyvrh-connection-0.2.8.drv': 11 dependencies couldn't be built
cannot build derivation '/nix/store/793ymrxpa1nmc0h9qbyb705hzaqpc6aa-yaml-0.8.32.drv': 15 dependencies couldn't be built
cannot build derivation '/nix/store/qcpay92v35qx4jgpckq7krz513ah3sdv-http-client-tls-0.3.5.3.drv': 13 dependencies couldn't be built
cannot build derivation '/nix/store/a1snaj0bpwzl00skil0a6xz97i0mkilf-pandoc-2.2.1.drv': 40 dependencies couldn't be built
cannot build derivation '/nix/store/2r6n4ij8p62869sx23fkhd6w9jy9nmap-rdma-core-20.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lbbwcz4vcny5s1byc5r3q09078ww6db6-openmpi-3.1.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/84m4d732z0bljyk7n92z56km3hqvzxwz-scalapack-2.0.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/1xwavjqb73zdv68jdca499nkdvfh1lr6-mxnet-1.2.1.drv', '/nix/store/84m4d732z0bljyk7n92z56km3hqvzxwz-scalapack-2.0.2.drv', '/nix/store/vnlhzv942a7wvc3xy3n7djpp5n33qw1l-gmsh-4.0.2.drv' failed
```
</p></details>

markuskowa commented:
This turns out to be more work than though. 
- `liblapack` can be removed from shogun. It's not used anymore. Doesn't solve the aarch64 build failure though (?).
- `scalapack`  has never been built successfully on darwin (set `platforms.linux` should be fine here).

matthewbauer committed:
shogun: remove liblapack

not needed anymore

GrahamcOfBorg commented:
<!--REQUEST_ID=373b0b43-8f8e-4ba8-b6ef-0db67f8da6c2-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=0ed48b15-5457-463b-88e1-b52ec2ef9908)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
./Main/ColPackHeaders.h:61:11: fatal error: 'omp.h' file not found
        #include <omp.h>
                 ^~~~~~~
1 error generated.
make[1]: *** [Makefile:1489: SampleDrivers/Basic/color_bipartite_graph_using_BipartiteGraphBicoloringInterface.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-ColPack-1.0.10.drv-0/source'
make: *** [Makefile:859: all] Error 2
builder for '/nix/store/2qdwr8gif8qixqvn5xfh0y6hvqg2q4f1-ColPack-1.0.10.drv' failed with exit code 2
cannot build derivation '/nix/store/gf67ngnfckz7y2xk345vbhqwhlsc4biv-shogun-6.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/gf67ngnfckz7y2xk345vbhqwhlsc4biv-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=373b0b43-8f8e-4ba8-b6ef-0db67f8da6c2-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=654fb8bf-486a-4bb9-89ee-fea60df1c568)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parallel.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/Parameter.cpp.o
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o
/build/ccIHndXN.s: Assembler messages:
/build/ccIHndXN.s:3821: Error: unknown mnemonic `pause' -- `pause'
make[2]: *** [src/shogun/CMakeFiles/libshogun.dir/build.make:1030: src/shogun/CMakeFiles/libshogun.dir/base/SGObject.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: src/shogun/CMakeFiles/libshogun.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/yfwy7ndnk7viw6fwlahmfycjm81w883d-shogun-6.0.0.drv' failed with exit code 2
error: build of '/nix/store/yfwy7ndnk7viw6fwlahmfycjm81w883d-shogun-6.0.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=373b0b43-8f8e-4ba8-b6ef-0db67f8da6c2-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=8cb65a8d-6b44-4c67-92a4-471e3d019004)

Attempted: liblapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/library_map
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/library_mldatahdf5
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/modelselection_apply_parameter_tree
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/modelselection_grid_search_string_kernel
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/evaluation_cross_validation_mkl_weight_storage
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/classifier_multiclass_ecoc_discriminant
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/lib
patching script interpreter paths in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0
checking for references to /build in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0...
```
</p></details>

markuskowa committed:
scalapack: set platform to linux

markuskowa committed:
colpack: set platform to linux

markuskowa committed:
Merge branch 'master' into atlas-remove

GrahamcOfBorg commented:
<!--REQUEST_ID=7cfcb212-e6d9-43dd-ad93-2cf8bf077923-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=14236ebd-fe80-425a-90a3-de5c19ebba71)

Attempted: liblapack

The following builds were skipped because they don't evaluate on x86_64-darwin: colpack, scalapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/d4qsxly2syvlji6likwl53yxb1b6j5bw-liblapack-3.8.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=7cfcb212-e6d9-43dd-ad93-2cf8bf077923-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=1fa37d3c-1441-4845-96ff-a76831813da2)

Attempted: colpack, liblapack, scalapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/parameter_set_from_parameters
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/lib/python2.7/site-packages/_modshogun.so
shrinking /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/lib/libshogun.so.18.0
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/lib
patching script interpreter paths in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0
checking for references to /build in /nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0...
cannot link '/nix/store/.links/10hani3vqnr0g1ckdaz7lfpg354afd0hcrvbdd5d9d5fn49l8ds0' to '/nix/store/byylcimd0g2lvg51ks58fn2h11c5mvk4-shogun-6.0.0/share/shogun/examples/libshogun/streaming_onlineliblinear_dense': No space left on device
copying path '/nix/store/ydl44v6sxijhrd63xnq9gswkqz52jn0p-scalapack-2.0.2' from 'https://cache.nixos.org'...
warning: unable to download 'https://cache.nixos.org/nar/1f63x497xk4z3l3xg1j1z1hb7nxrhdgijdz6c1xmailc6kk289rn.nar.xz': Error in the HTTP2 framing layer (16); retrying in 255 ms
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=7cfcb212-e6d9-43dd-ad93-2cf8bf077923-->
Timed out, unknown build status on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49776&attempt_id=49ee0452-9a10-4279-82b4-080bb8b8481e)

Attempted: colpack, liblapack, scalapack, shogun

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/by5p7vlw6p1hsixyliy85dim5r8xqzq5-conduit-1.3.1.drv': 14 dependencies couldn't be built
cannot build derivation '/nix/store/q3pv07v48jyh8qzgpm1r8wjv6yc5va57-tls-1.4.1.drv': 17 dependencies couldn't be built
cannot build derivation '/nix/store/x94qhdgjjn6pi5zhcnbxp6csmndvyvrh-connection-0.2.8.drv': 11 dependencies couldn't be built
cannot build derivation '/nix/store/793ymrxpa1nmc0h9qbyb705hzaqpc6aa-yaml-0.8.32.drv': 15 dependencies couldn't be built
cannot build derivation '/nix/store/qcpay92v35qx4jgpckq7krz513ah3sdv-http-client-tls-0.3.5.3.drv': 13 dependencies couldn't be built
cannot build derivation '/nix/store/a1snaj0bpwzl00skil0a6xz97i0mkilf-pandoc-2.2.1.drv': 40 dependencies couldn't be built
cannot build derivation '/nix/store/2r6n4ij8p62869sx23fkhd6w9jy9nmap-rdma-core-20.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lbbwcz4vcny5s1byc5r3q09078ww6db6-openmpi-3.1.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/84m4d732z0bljyk7n92z56km3hqvzxwz-scalapack-2.0.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/84m4d732z0bljyk7n92z56km3hqvzxwz-scalapack-2.0.2.drv', '/nix/store/yfwy7ndnk7viw6fwlahmfycjm81w883d-shogun-6.0.0.drv' failed
```
</p></details>
